src/lib.rs

```diff
@@ -978,9 +978,7 @@ pub struct LocalVariable {
     pub name: Option<String>,
     /// The type of this variable.
     pub ty: Handle<Type>,
-    /// Initial value for this variable.
-    ///
-    /// Expression handle lives in const_expressions
+    /// Initial value for this variable. Must be a const-expression.
```

Comment:
The term "const-expression" isn't defined anywhere in this file. I assume it should say "evaluated expression"?

There's no validation of this constraint.

Why did this move? It seems odd to require it to be an evaluated expression (I think that's the terminology we settled on, right?) but have it live in the function's expression arena.

Reply:
> The term "const-expression" isn't defined anywhere in this file. I assume it should say "evaluated expression"?

`const-expression` is what the spec calls them but we have defined them as "constant expressions" at the top of the file in https://github.com/gfx-rs/naga/commit/c7490b9832468ca7d1539978091c4769fe148a52; the "use LocalVariable init" commit originally predated that commit and updates like these get lost in rebases.

Similarly to the comment I left on the discussion above, when we add full support for override expressions, this will also be allowed to refer to override expressions which will include the other variants of "constant expressions".

> There's no validation of this constraint.

Added in https://github.com/gfx-rs/naga/pull/2309/commits/1c781270415a6ecfe808b98d6f48a525417343fd.

> Why did this move?

We would have had to copy the evaluated expressions into the `const_expressions` arena which didn't seem worth doing.